tests/aggregation_regress/tests.py

```diff
@@ -1523,6 +1524,25 @@ class DistinctAggregate(Aggregate):
             allow_distinct = True
         DistinctAggregate('foo', distinct=True)
 
+    def test_groupby_after_aggregation_and_subquery(self):
+        """
+        Regression test for #32152.
+        Groupby column aliases should be used when present in the select statement.
+        """
+        self.assertEqual(
+            Book.objects.all().annotate(
+                pub_year=TruncYear('pubdate')
```

Comment:
I think you want `ExtractYear` here, `TruncYear(DateFieldField) -> DateTimeField` while `ExtractYear(DateFieldField) -> IntegerField`.

Reply:
You are correct, I've made the change. Interesting that it works on all the other databases...

Reply:
Yeah I was surprised other backends didn't complain as well